packages/base-manager/src/manager-base.ts

```diff
@@ -333,11 +328,13 @@ export abstract class ManagerBase implements IWidgetManager {
    * Close all widgets and empty the widget state.
    * @return Promise that resolves when the widget state is cleared.
    */
-  clear_state(): Promise<void> {
-    return resolvePromisesDict(this._models).then(models => {
-      Object.keys(models).forEach(id => models[id].close());
-      this._models = Object.create(null);
-    });
+  async clear_state(): Promise<void> {
+    await Promise.all(
+      Object.keys(this._models).map(async id =>
+        (await this._models[id]).close()
+      )
+    );
```

Comment:
Note: This has a small difference in nuance, which I think opens up a new corner case. If one of the unresolved models still being created rely on the models that it references not being closed, this change might cause it to throw an error where it previously did not. To clarify:
- The old method waited for all models currently in `this._models` to resolve, then closed all of them in one go (synchronously).
- The new method will close each model as soon as they resolve. The ID will still be present in the map, but the model it points to is closed.

That said, I don't really see `clear_state` being used anywhere, so it us likely to actually affect anything.